micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -776,6 +778,19 @@ public Config commonTags(String... tags) {
          * @return This configuration instance.
          */
         public synchronized Config meterFilter(MeterFilter filter) {
+            if (!meterMap.isEmpty()) {
+                InternalLogger logger = InternalLoggerFactory.getInstance(MeterRegistry.this.getClass());
+                String baseMessage = "A MeterFilter is being configured after a Meter has been registered to this registry. This is unexpected and should be corrected so all MeterFilters are configured before any Meters are registered. If that is not possible or you have a use case where it should be allowed, contact the Micrometer maintainers.";
```

Comment:
"This is unexpected" begs the question why, which the log message doesn't explain now, but it is already a long log message, so I was avoiding expanding more. If we do end up including a link to a GitHub issue, maybe that will alleviate this problem as we can expand more easily (and update the explanation) in a GitHub issue than we can in a log message.